infra/staging/main.tf

```diff
@@ -0,0 +1,112 @@
+###############
+## Variables ##
+###############
+
+variable "project_id" {
+  type = string
+  default = "call-home-staging"
+}
+
+variable "location" {
+  type = string
+  default = "asia-southeast1"
+}
+
+variable "db_name" {
+  type = string
+  default = "strapi"
+}
+
+variable "db_instance_name" {
+  type = string
+  default = "strapi-db-instance"
+}
+
+variable "db_username" {
+  type = string
+  sensitive = true
+}
+
+variable "db_password" {
+  type = string
+  sensitive = true
+}
+
+######################
+## Project Settings ##
+######################
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+// Enable Cloud Build API and grant its service account App Engine and Service Account User
+// permissions to allow developers to build and deploy applications to App Engine using gcloud SDK.
+// See: https://cloud.google.com/build/docs/deploying-builds/deploy-appengine
+resource "google_project_service" "cloud_build_api" {
+  project = data.google_project.project.project_id
+  service = "cloudbuild.googleapis.com"
+}
+
+data "google_service_account" "cloud_build_service_account" {
+  account_id = google_project_service.cloud_build_api.id
```

Comment:
Oh interesting, I didn't know you could reference the cloud build account like this.